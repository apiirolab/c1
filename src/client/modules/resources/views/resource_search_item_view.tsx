import * as React from 'react';

import { List } from 'semantic-ui-react';

import { context } from 'client/config/context';

type Props = {
  searchItem: Corpix.Entities.Search;
};

export const SearchView: React.SFC<Props> = ({ searchItem }) => (
  <List.Item>
    <List.Icon name="file outline" verticalAlign="middle" />
    <List.Content>
      <List.Header as="a">{searchItem.category}</List.Header>
      <List.Description as="a">
        Submitted by {searchItem.owner} &middot; {searchItem.title} &middot;{' '}
        {context.Ui.relativeDate(searchItem.date)}
      </List.Description>
    </List.Content>
  </List.Item>
);
