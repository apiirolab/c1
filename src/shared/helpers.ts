interface Indexable<T> {
  [index: string]: T;
}

interface Group<T> {
  key: string;
  values: T[];
}

export function groupBy<T>(xs: T[], key: string): Indexable<T[]> {
  return xs.reduce(function(previous: any, current: Indexable<any>) {
    // Indexable<Array<Indexable<T>>>
    (previous[current[key]] = previous[current[key]] || []).push(current);
    return previous;
  }, {});
}

export function groupByArray<T>(xs: T[], key: string | Function): Array<Group<T>> {
  return xs.reduce(function(previous, current: Indexable<any>) {
    let v = key instanceof Function ? key(current) : current[key];
    let el = previous.find(r => r && r.key === v);
    if (el) {
      el.values.push(current);
    } else {
      previous.push({
        key: v,
        values: [current]
      });
    }
    return previous;
  }, []);
}
