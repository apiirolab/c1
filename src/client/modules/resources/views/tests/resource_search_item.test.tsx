import * as React from 'react';

import { create as render } from 'react-test-renderer';

import { create } from 'shared/test_data';
import { SearchView } from '../resource_search_item_view';

describe('Resource', () => {
  storyOf(
    'Search',
    {
      componentWithData(searchItem: Corpix.Entities.Search) {
        return <SearchView searchItem={searchItem} />;
      }
    },
    data => {
      it('renders default view', () => {
        const root = render(data.componentWithData(create.searchItem({ category: 'Travel Request Form', owner: 'Tomas Trescak', title: 'International Travel Request', date: new Date(2010, 1, 2,) })));
        expect(root).toMatchSnapshot();
      });
    }
  );
})