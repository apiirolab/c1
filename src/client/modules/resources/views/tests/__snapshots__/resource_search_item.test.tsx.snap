// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Search renders default view 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <i
    aria-hidden="true"
    className="file outline icon middle aligned"
  />
  <div
    className="content"
  >
    <a
      className="header"
    >
      Travel Request Form
    </a>
    <a
      className="description"
    >
      Submitted by 
      Tomas Trescak
       · 
      International Travel Request
       ·
       
      in 9 years
    </a>
  </div>
</div>
`;
